projects/fsastorefrontlib/src/core/claim/guards/claim-confirmation-guard.ts

```diff
@@ -5,25 +5,38 @@ import {
   GlobalMessageType,
   RoutingService,
 } from '@spartacus/core';
-import { Observable } from 'rxjs';
+import { combineLatest, Observable } from 'rxjs';
 import { ClaimStatus } from '../../../occ/occ-models/occ.models';
-import { map } from 'rxjs/operators';
-import { UserRequestService } from '../../user-request/facade';
+import { filter, map } from 'rxjs/operators';
+import { ClaimService } from '../../my-account';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ClaimConfirmationGuard implements CanActivate {
   constructor(
-    protected userRequestService: UserRequestService,
     protected routingService: RoutingService,
-    protected globalMessageService: GlobalMessageService
+    protected globalMessageService: GlobalMessageService,
+    protected claimService: ClaimService
   ) {}
 
+  private readonly claimConfirmationStatuses = [
+    ClaimStatus.APPROVED,
+    ClaimStatus.SUBMITTED,
+    ClaimStatus.PROCESSING,
+  ];
+
   canActivate(): Observable<boolean | UrlTree> {
-    return this.userRequestService.getUserRequest().pipe(
-      map(claim => {
-        if (claim.requestStatus === ClaimStatus.SUBMITTED) {
+    return combineLatest([
+      this.claimService.getLoaded(),
+      this.claimService.getCurrentClaim(),
+    ]).pipe(
+      filter(([loaded, _]) => loaded),
+      map(([_, claim]) => {
+        const isClaimFinished = this.claimConfirmationStatuses.includes(
+          claim.claimStatus
+        );
+        if (isClaimFinished) {
```

Comment:
Maybe we could move this condition to filter:

 filter(([loaded, claim]) => loaded && this.claimConfirmationStatuses.includes(
          claim.claimStatus
        ))

Reply:
Then return statement wont be executed. Customer will be stuck on page .

Reply:
You are right, I didn't see the rest of the code :)